client/src/components/_common/TextCopyField/TextCopyField.jsx

```diff
@@ -30,44 +30,74 @@ const TextCopyField = ({ value, placeholder }) => {
 
   return (
     <div className="input-group">
-      <div className="input-group-prepend">
-        <CopyToClipboard text={value}>
-          <Button
-            className={styles['copy-button']}
-            // RFE: Avoid manual JS ↔ CSS sync of transition duration by using:
-            //      - `data-attribute` and `attr()` (pending browser support)
-            //      - PostCSS and JSON variables (pending greater need for this)
-            onClick={onCopy}
-            disabled={isEmpty}
-            type="secondary"
-            size="medium"
-            iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
-          >
-            Copy
-          </Button>
-        </CopyToClipboard>
-      </div>
-      <input
-        type="text"
-        onChange={onChange}
-        value={value}
-        className={`form-control ${styles.input}`}
-        placeholder={placeholder}
-        data-testid="input"
-        readOnly
-      />
+      {renderType === 'textarea' ? (
+        <>
+          <textarea
+            onChange={onChange}
+            value={value}
+            className={`${styles.textarea}`}
+            placeholder={placeholder}
+            data-testid="textarea"
+            readOnly
+          />
+          <div className="text-right mt-3">
+            <CopyToClipboard text={value}>
+              <Button
+                onClick={onCopy}
+                disabled={isEmpty}
+                type="secondary"
+                size="medium"
+                iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
+              >
+                Copy
+              </Button>
+            </CopyToClipboard>
+          </div>
+        </>
+      ) : (
+        <>
+          <div className="input-group-prepend">
+            <CopyToClipboard text={value}>
+              <Button
+                className={styles['copy-button']}
+                // RFE: Avoid manual JS ↔ CSS sync of transition duration by using:
+                //      - `data-attribute` and `attr()` (pending browser support)
+                //      - PostCSS and JSON variables (pending greater need for this)
+                onClick={onCopy}
+                disabled={isEmpty}
+                type="secondary"
+                size="medium"
+                iconNameBefore={isCopied ? 'approved-reverse' : 'link'}
+              >
+                Copy
+              </Button>
+            </CopyToClipboard>
+          </div>
+          <input
+            type="text"
+            onChange={onChange}
+            value={value}
+            className={`form-control ${styles.input}`}
+            placeholder={placeholder}
+            data-testid="input"
+            readOnly
+          />
+        </>
+      )}
     </div>
   );
 };
 
 TextCopyField.propTypes = {
   value: PropTypes.string,
   placeholder: PropTypes.string,
+  renderType: PropTypes.oneOf(['input', 'textarea']),
```

Comment:
Excellent use of a prop for this.

What thoughts do you have on the name `renderType`? Do you anticipate any other kind of `type` prop? Could any part of this component, besides the field, be rendered differently via prop in the future?

_Naming is tough. Being too specific can lead to verbosity. Being too vague can lead to overloading a prop._